src/coq/logicutils/typesandequality/inference.ml

```diff
@@ -10,12 +10,14 @@ open Declarations
 (* Safely infer the WHNF type of a term, updating the evar map *)
 let infer_type env sigma term =
   let eterm = EConstr.of_constr term in
-  Typing.type_of ~refresh:true env sigma eterm
+  let sigma, typ = Typing.type_of ~refresh:true env sigma eterm in
+  sigma, EConstr.to_constr sigma typ
```

Comment:
Why is this needed? Typing.tpe_of was already returning econstr in 8.9 AFAICT https://github.com/coq/coq/blob/v8.9/pretyping/typing.mli#L25

Reply:
> Why is this needed? Typing.tpe_of was already returning econstr in 8.9 AFAICT https://github.com/coq/coq/blob/v8.9/pretyping/typing.mli#L25

I just reverted that back to the way it was in v8.9 of the plugin (in the last commit), I had made a change that was causing a type error